src/azure-cli/azure/cli/command_modules/acs/tests/latest/custom_preparers.py

```diff
@@ -31,3 +32,87 @@ def __init__(
+                             resource_group_parameter_name, resource_group_key,
+                             dev_setting_name, random_name_length, key)
+
+        # use environment variable to modify the default value of location
+        self.dev_setting_location = os.environ.get(dev_setting_location, None)
+        if self.dev_setting_location:
+            self.location = self.dev_setting_location
+        else:
+            self.dev_setting_location = location
+
+        # get address_prefixes
+        # custom address_prefixes to avoid conflict with aks cluster/service cidr
+        self.address_prefixes = address_prefixes
+        self.address_prefixes_parameter_name = address_prefixes_parameter_name
+
+    def create_resource(self, name, **kwargs):
+        if self.dev_setting_name:
+            self.test_class_instance.kwargs[self.key] = name
+            return {
+                self.parameter_name: self.dev_setting_name,
+            }
+
+        tags = {
+            'product': 'azurecli',
+            'cause': 'automation',
+            'date': datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ')
+        }
+        if 'ENV_JOB_NAME' in os.environ:
+            tags['job'] = os.environ['ENV_JOB_NAME']
+        tags = ' '.join(
+            ['{}={}'.format(key, value) for key, value in tags.items()])
+        template = 'az network vnet create --resource-group {} --location {} --name {} --subnet-name default --address-prefixes {} --tag ' + tags
+        self._update_address_prefixes(**kwargs)
+        self.live_only_execute(
+            self.cli_ctx,
+            template.format(self._get_resource_group(**kwargs), self.location,
+                            name, self.address_prefixes))
+
+        self.test_class_instance.kwargs[self.key] = name
+        return {self.parameter_name: name}
+
+    def remove_resource(self, name, **kwargs):
+        if not self.dev_setting_name:
+            from msrestazure.azure_exceptions import CloudError
+            from azure.core.exceptions import HttpResponseError
+            try:
+                self.live_only_execute(
+                    self.cli_ctx,
+                    'az network vnet delete --name {} --resource-group {}'.
+                    format(name, self._get_resource_group(**kwargs)))
+            except CloudError:
+                # deprecated since the SDK that network commands relies on has been migrated to track 2
+                # deletion of vnet may fail as service could create subresources like IPConfig. We could rely on the deletion of resource group to delete the vnet.
+                pass
+            except HttpResponseError as ex:
+                # new excetption class adopted by track 2 SDK
+                # network resources are still used by other resources such as vm, and there is no clean way to delete these dependencies, rely on delete rg.
+                if "(InUseSubnetCannotBeDeleted)" in ex.message:
```

Comment:
is there any error code we can use instead of the text matching?

Reply:
change to pass and leave comments in commit [a7e9704](https://github.com/Azure/azure-cli/pull/18168/commits/a7e97046c8cdd779ef0fb717427ca25fffbca3bb).